Skip when no region is set

jnhmcknight opened the pull request:
When in development, Dynaconf does not necessarily need to have credentials sourced, but when loading this extension without credentials, Dynaconf will error out and not proceed past the fact that there are no credentials. I believe this should fail gracefully instead, and skip using this loader.

This outputs the exception, and continues on with Dynaconf loading. Thereby allowing you to know what happened without actually blocking anything.

jnhmcknight committed:
feat: Allow initialization to continue without crashing if no region is set

jperras commented:
@jnhmcknight thanks for catching this. I'm going to make a small change to the PR to utilize the `silent: bool` parameter that is already present and used for other error types in the loader; that will at least make the handling a bit more consistent.

jperras committed:
refactor: Unify exception handling style

Makes things a bit more in line with what Dynaconf expects regarding the
use of the ``silent`` parameter. Additionally, we unrolled a large
try/except block that really only needed to cover a single statement.

jperras commented:
@jnhmcknight I updated the PR. I reworded your commit and force pushed the change so that it follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) guidelines, so please be aware of this if you pull the branch.
